src/ppx/graphics_util.cpp

```diff
@@ -1510,5 +1683,58 @@ Result CreateMeshFromFile(
     return ppx::SUCCESS;
 }
 
+// -------------------------------------------------------------------------------------------------
+
+Result LoadFramesFromRawVideo(
+    const std::filesystem::path&    path,
+    grfx::Format                    format,
+    uint32_t                        width,
+    uint32_t                        height,
+    std::vector<std::vector<char>>* pFrames)
+{
+    PPX_ASSERT_NULL_ARG(pFrames);
+
+    const grfx::FormatDesc* formatDesc = grfx::GetFormatDescription(format);
+    if (formatDesc == nullptr) {
+        PPX_LOG_ERROR("Failed to fetch information for texture format " << format);
+        return ppx::ERROR_FAILED;
+    }
+
+    std::optional<grfx::FormatPlaneDesc> formatPlanes = grfx::GetFormatPlaneDescription(format);
+
+    uint32_t frameSizeInBytes = 0;
+    if (formatPlanes.has_value()) {
+        frameSizeInBytes = GetPlanarImageSizeInBytes(*formatDesc, *formatPlanes, width, height);
+    }
+    else {
+        frameSizeInBytes = formatDesc->bytesPerTexel * width * height;
+    }
+
+    ppx::fs::File file;
+    if (!file.Open(path)) {
+        PPX_ASSERT_MSG(false, "Cannot open the file!");
+        return ppx::ERROR_FAILED;
+    }
+    const size_t size = file.GetLength();
+
+    if (size % frameSizeInBytes != 0) {
+        PPX_LOG_WARN("There may be partial frames in the raw video file at " << path << "; loading as much as possible.");
+    }
+
+    std::vector<char> buffer(frameSizeInBytes);
+    size_t            totalRead = 0;
+    while (totalRead + frameSizeInBytes <= size) {
```

Comment:
shall the last partial frame be loaded and unavailable bits be set to zero?
Or shall this be a hard error?
In what cases do we expect this to happen?

Reply:
I really don't expect it in any case, for someone to have an incomplete video file. I'm more open to the first option than the second, though, as I wouldn't want to prevent someone from using such a file in a benchmark. In almost all cases, I expect it to be a single frame "video" that's being loaded, as a snapshot of a camera feed.